### DISCUSSION TITLE
Updating multiple fields by relationship

### DISCUSSION THREAD

**Discussion by ragokan** [prisma/prisma#12084] [Q&A] [answered]
Hello, is there any way to update multiple fields by relationship such as adding all of them a message?

Currently this is what I do

1: for loop in users
2: prisma.user.update -> where id: user.id -> data: message: create: id: X

what I want to do
prisma.user.updateMany -> data: message: create: id: X

**Comment by austincrim** [comment]
Hey @ragokan 👋 ,

Could you share your `schema.prisma` and a code snippet of what you want to accomplish? I just want to make sure I understand your question correctly.

**Reply by ragokan** [reply]
Hello.

What I want to do is
```ts
await prisma.user.updateMany({data:{ messages: {create: { text: "This message is added to every user!"  }} }})
```

What I am currently doing is
```ts
    // get all the users
    const users = await prisma.user.findMany();
    
    // loop over the users
    for (let index = 0; index < users.length; index++) {
      const user = users[index];
      
      // update each user one by one
      await prisma.user.update({
        where: { id: user.id },
        data: { messages: { create: { text: "This message is added to every user!" } } },
      });
    }
```

example schema is;
```prisma
model User {
  id        Int      @id @default(autoincrement())
  name      String
  messages     Message[]
}

model Message {
  id        Int      @id @default(autoincrement())
  text     String   @db.VarChar(255)
  author    User?    @relation(fields: [authorId], references: [id])
  authorId  Int?
}
```

**Comment by austincrim** [comment, marked as answer]
This is not possible yet with the Prisma Client (see #3143). Feel free to add your use case and +1 to that existing issue to help our engineering teams prioritize!

In the meantime, you could use a [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#executeraw) to accomplish this or stick with your looping workaround.

**Reply by ragokan** [reply]
Oh, okay. Thank you!

**Reply by Noitidart** [reply]
Thanks @austincrim, can you please give an example of how to do this with `executeRaw`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
